### PR TITLE
Terminal rewards

### PR DESCRIPTION
**Pull request by DanielHettegger**
Merge Terminal rewards into main. Merge after #86 

[x] Pytests passing
[x] Linter applied.

### CODE REVIEW

**Comment by moratodpg** [comment]
Thanks, @DanielHettegger!

I do not see any issue with the implementation. 💯 

Let's perhaps include the final modifications applied to #86.

Also, I would not merge this branch into "main" before running some simulations (heuristics & MARL). From a few preliminary rollouts, I see the terminal cost is still very high.

File: imp_act/environments/road_env.py [hunk 1]
```diff
@@ -121,9 +145,26 @@ def get_initial_state(self):
...
+            )
+            reward = (
+                self.state_action_reward[action][self.state] * forced_repair_interest
+            )
+            self.forced_repair_interest_counter = 0
+        else:
+            reward = self.state_action_reward[action][self.state]
+        return reward
+
+    def get_terminal_reward(self):
```
Comment (moratodpg) [inline imp_act/environments/road_env.py:165]: Are we sure we would like to charge terminal costs based on beliefs? I suggested this option, but is this okay with our reward formulation? Let's perhaps all agree.

**Comment by DanielHettegger** [comment]
Thanks Pablo! Should we tune the final rewards somehow before merging or is it okay like this?

**Comment by moratodpg** [comment]
We can leave it as is IMO. 
In any case, we can always modify the values from the config later on. :)

**Pull request merged**